xbmc/BackgroundInfoLoader.cpp

```diff
@@ -90,6 +90,9 @@ void CBackgroundInfoLoader::Run()
     m_bIsLoading = false;
     CLog::Log(LOGERROR, "{} - Unhandled exception", __FUNCTION__);
   }
+
+  m_pVecItems = nullptr;
```

Comment:
Probably create a reset method you can reuse to set all the members to default?
Same seems to be done in stop thread.